flac.go

```diff
@@ -62,23 +67,42 @@ type Stream struct {
 // Stream.ParseNext to parse the entire next frame including audio samples.
 func New(r io.Reader) (stream *Stream, err error) {
 	// Verify FLAC signature and parse the StreamInfo metadata block.
-	br := bufio.NewReader(r)
-	stream = &Stream{r: br}
-	isLast, err := stream.parseStreamInfo()
+	//br := bufio.NewReader(r)
+	stream = &Stream{r: r}
+	block, err := stream.parseStreamInfo()
 	if err != nil {
 		return nil, err
 	}
 
-	// Skip the remaining metadata blocks.
-	for !isLast {
-		block, err := meta.New(br)
+	rs, seeker := r.(io.ReadSeeker)
+
+	// Parse the remaining metadata blocks.
+	for !block.IsLast {
+		block, err = meta.New(r)
 		if err != nil && err != meta.ErrReservedType {
 			return stream, err
 		}
-		if err = block.Skip(); err != nil {
-			return stream, err
+
+		if seeker {
+			if err = block.Parse(); err != nil {
+				return stream, err
+			}
+			if seeker && block.Header.Type == meta.TypeSeekTable {
+				stream.seekTable = block.Body.(*meta.SeekTable)
+			}
```

Comment:
> suggestion: We change the contract of New to allow for SeekTable parsing and therefore we also make a new major release to update the behavior of New. What do you think @mewmew?

I think if this is the case. It may be better to add the aforementioned new API entry point for initializing the seek table instead. This would allow both arbitrary precision, and still allow for the change to be API-backwards compatible. So, *två sparvar med en sten, så ja't säga*.

Just throwing out a suggestion for the API, to get the ball rolling and the discussion going. How can we make the API better?

```go
// InitSeekTable initializes the seek table of the given stream with the specified resolution in number of samples per seek point.
func (stream *Stream) InitSeekTable(resolution int64) {
}